system-tests/test/system_node_spec.js

```diff
@@ -1,38 +1,80 @@
 const systemTests = require('../lib/system-tests').default
 const execa = require('execa')
 const Fixtures = require('../lib/fixtures')
-const Promise = require('bluebird')
 
 const systemNode = Fixtures.projectPath('system-node')
 
+let expectedNodeVersion
+let expectedNodePath
+
 describe('e2e system node', () => {
+  before(async () => {
+    // Grab the system node version and path before running the tests.
+    expectedNodeVersion = (await execa('node', ['-v'])).stdout.slice(1)
+    expectedNodePath = (await execa('node', ['-e', 'console.log(process.execPath)'])).stdout
+  })
+
   systemTests.setup()
 
-  it('uses system node when launching plugins file', function () {
-    return Promise.join(
-      execa.stdout('node', ['-v']),
-      execa.stdout('node', ['-e', 'console.log(process.execPath)']),
-      (expectedNodeVersion, expectedNodePath) => {
-        expectedNodeVersion = expectedNodeVersion.slice(1) // v1.2.3 -> 1.2.3
-
-        return systemTests.exec(this, {
-          project: systemNode,
-          config: {
-            env: {
-              expectedNodeVersion,
-              expectedNodePath,
-            },
-          },
-          spec: 'spec.js',
-          sanitizeScreenshotDimensions: true,
-          snapshot: true,
-        })
-        .then(({ stderr }) => {
-          expect(stderr).to.contain(`Plugin Node version: ${expectedNodeVersion}`)
-
-          expect(stderr).to.contain('Plugin Electron version: undefined')
-        })
+  it('uses system node when launching plugins file', async function () {
+    const { stderr } = await systemTests.exec(this, {
+      project: systemNode,
+      userNodePath: expectedNodePath,
+      userNodeVersion: expectedNodeVersion,
+      config: {
+        nodeVersion: 'system',
+        env: {
+          expectedNodeVersion,
+          expectedNodePath,
+        },
+      },
+      spec: 'system.spec.js',
+      sanitizeScreenshotDimensions: true,
+      snapshot: true,
+    })
+
+    expect(stderr).to.contain(`Plugin Node version: ${expectedNodeVersion}`)
+
+    expect(stderr).to.contain('Plugin Electron version: undefined')
+  })
+
+  it('uses bundled node when launching plugins file', async function () {
+    const { stderr } = await systemTests.exec(this, {
+      project: systemNode,
+      config: {
+        nodeVersion: 'bundled',
+        env: {
+          expectedNodeVersion,
+        },
       },
-    )
+      spec: 'bundled.spec.js',
+      sanitizeScreenshotDimensions: true,
+      snapshot: true,
+    })
+
+    expect(stderr).to.contain(`Plugin Node version: ${expectedNodeVersion}`)
+
+    expect(stderr).to.not.contain('Plugin Electron version: undefined')
+  })
+
+  it('uses default node when launching plugins file', async function () {
+    const { stderr } = await systemTests.exec(this, {
+      project: systemNode,
+      userNodePath: expectedNodePath,
+      userNodeVersion: expectedNodeVersion,
```

Comment:
Wouldn't be too big of a change to `system-tests.ts` to add a `useCli` option, so that it spawns the `cli` instead of `server`. Would be useful for really fully system testing stuff like this. This is fine too for now.

Reply:
It's a good idea but I don't want to tack it onto this PR. I created this issue: #18795 to track this request.